great_expectations/core/util.py

```diff
@@ -430,6 +434,29 @@ def blob(self):
         return self._blob
 
 
+class GCSUrl:
+    """
+    Parses a Google Cloud Storage URL into its separate components
+    Format: gs://<BUCKET_OR_NAME>/<BLOB>
+    """
+
+    def __init__(self, url: str):
+        search = re.search(r"^gs://([^/]+)/(.+)$", url)
+        assert (
+            search is not None
+        ), "The provided URL does not adhere to the format specified by the GCS SDK (gs://<BUCKET_OR_NAME>/<BLOB>)"
```

Comment:
Invariant that should theoretically never be violated but a good practice given our usage of regular expressions

Reply:
Thank you for this :)